pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -110,9 +111,21 @@ func prepareSubpathTarget(mounter mount.Interface, subpath Subpath) (bool, strin
 		notMount = true
 	}
 	if !notMount {
-		// It's already mounted
-		klog.V(5).Infof("Skipping bind-mounting subpath %s: already mounted", bindPathTarget)
-		return true, bindPathTarget, nil
+		linuxHostUtil := hostutil.NewHostUtil()
+		mntInfo, err := linuxHostUtil.FindMountInfo(bindPathTarget)
+		if err != nil {
+			return false, "", fmt.Errorf("error calling findMountInfo for %s: %s", bindPathTarget, err)
+		}
+		if mntInfo.Root != subpath.Path {
+			// It's already mounted but not what we want, unmount it
+			err = mounter.Unmount(bindPathTarget)
```

Comment:
is it safe to unmount the bindPathTarget?

Reply:
Naive question, can we use symlink instead of bind mount? So after atomic_writer remove the old timestamped dir and create the new one, the subpath target is still valid to mount.

Reply:
I have read https://kubernetes.io/blog/2018/04/04/fixing-subpath-volume-vulnerability/ . It explains why using bind mount.

Reply:
I think unmounting the subpath here is safe. It is a per-container bind mount and the container is not running here. wdyt @jsafrane ?